src/run.ts

```diff
@@ -81,7 +81,7 @@ export const run = async (
         'baseCoverage',
         dataCollector,
         async (skip) => {
-            if (!isSwitched) {
+            if (!isSwitched && options.coverageFile === options.baseCoverageFile) {
```

Comment:
I think, that most suitable logic would be:
```ts
if (!isSwitched && options.baseCoverageFile === undefined)
```
Because this logic handles these situations:
* When `baseCoverageFile` is not specified:
   Stage will run depending on result of switching to the base branch, because there is no reason to collect coverage information on the same branch twice.
* When `baseCoverageFiel` is specified:
   Stage will run always.

Reply:
This logic works for my case. What would happen though if `isSwitched` is true but `baseCoverageFile` is `undefined`, given [L88](https://github.com/ArtiomTr/jest-coverage-report-action/blob/237287b4b4c4a997c912977cfd823514f23abbe6/src/run.ts#L88 uses `baseCoverageFile`?

Perhaps undefined is handled in that func though or this is an impossible state.

Reply:
If `isSwitched` is true, then the condition that @ArtiomTr is proposing will evaluate to false, no matter what the value of `baseCoverageFile`, so I think it is the proper logic
```suggestion
            if (!isSwitched && options.baseCoverageFile === undefined) {
```